components/scream/src/diagnostics/dry_static_energy.cpp

```diff
@@ -63,27 +70,30 @@ void DryStaticEnergyDiagnostic::compute_diagnostic_impl()
   const Real surf_geopotential = 0.0;
 
   const int num_levs = m_num_levs;
-  view_1d dz("",npacks);
-  view_1d z_int("",npacks_p1);
-  view_1d z_mid("",npacks);
+  auto      dz       = m_dz;
```

Comment:
If you really wanted, you could store dz in `m_z_mid`. By the time you compute z_mid, you no longer need dz...